backend/utils.py

```diff
@@ -1,25 +1,101 @@
+"""Backend utilities"""
 import json
+import operator
+from datetime import datetime
+from functools import reduce
+from typing import Any, Dict, List, Tuple, Union
 
 
 def cnt(vals):
-  return len(set(vals))
+    """Count values"""
+    return len(set(vals))
 
 
 def commify(n):
-  return f'{n:,}'
+    """Those dirty commies"""
+    return f'{n:,}'
 
 
 def dump(o):
-  return json.dumps(o, indent=2)
+    """Return pretty printed json"""
+    return json.dumps(o, indent=2)
 
 
 def pdump(o):
-  print(dump(o))
+    """Prent pretty printed json"""
+    print(dump(o))
 
 
-class Bunch(object):    # dictionary to namespace, a la https://stackoverflow.com/a/2597440/1368860
-  def __init__(self, adict):
-    self.__dict__.update(adict)
+class Bunch(object):
+    """dictionary to namespace, a la https://stackoverflow.com/a/2597440/1368860"""
+    def __init__(self, adict):
+        """Init"""
+        self.__dict__.update(adict)
 
-  def to_dict(self):
-    return self.__dict__
+    def to_dict(self):
+        """Convert to dictionary"""
+        return self.__dict__
+
+
+def get_nested_from_dict(d: Dict, key_path: List):
```

Comment:
FYI: `get_nested_from_dict()` and `set_nested_in_dict()`
Helper functions for dealing with nested dictionaries. Used by `hierarchify_list_of_parent_kids()`.